File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
#include <stack>
#include <sstream>

class Solution {
public:
    string reverseWords(string s) {
        stack<string> st;
        stringstream ss(s);
        string word;
        
        while (ss >> word) {
            st.push(word);
        }
        
        string ans;
        while (!st.empty()) {
            ans += st.top();
            ans += ' ';
            st.pop();
        }
        
        if (!ans.empty()) {
            ans.pop_back();
        }
        
        return ans;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        stack = []
        ans = ""
        i = 0

        while i < len(s):
            if s[i] != " ":
                temp = ""
                while i < len(s) and s[i] != " ":
                    temp += s[i]
                    i += 1
                stack.append(temp)
            else:
                i += 1
        
        while stack:
            # ans += stack[-1]
            # ans += " "
            # stack.pop()
            ans += stack.pop() + " "

        return ans.strip()